src/lib/services/access-service.ts

```diff
@@ -300,7 +381,82 @@ export class AccessService {
     }
 
     async getRootRole(roleName: RoleName): Promise<IRole> {
-        const roles = await this.store.getRootRoles();
+        const roles = await this.roleStore.getRootRoles();
         return roles.find((r) => r.name === roleName);
     }
+
+    async getAllRoles(): Promise<ICustomRole[]> {
+        return this.roleStore.getAll();
+    }
+
+    async createRole(role: IRoleCreation): Promise<ICustomRole> {
+        await this.validateRole(role);
+        const baseRole = {
+            name: role.name,
+            description: role.description,
+            roleType: 'custom',
+        };
+        const rolePermissions = role.permissions;
+        const newRole = await this.roleStore.create(baseRole);
+        if (rolePermissions) {
+            this.store.addEnvironmentPermissionsToRole(
+                newRole.id,
+                rolePermissions,
+            );
+        }
+        return newRole;
+    }
+
+    async updateRole(role: IRoleUpdate): Promise<ICustomRole> {
+        await this.validateRole(role, role.id);
+        const baseRole = {
+            id: role.id,
+            name: role.name,
+            description: role.description,
+            roleType: 'custom',
+        };
+        const rolePermissions = role.permissions;
+        const newRole = await this.roleStore.update(baseRole);
+        if (rolePermissions) {
+            this.store.wipePermissionsFromRole(newRole.id);
+            this.store.addEnvironmentPermissionsToRole(
+                newRole.id,
+                rolePermissions,
+            );
+        }
+        return newRole;
+    }
+
+    async deleteRole(id: number): Promise<void> {
```

Comment:
Does this cascade to delete permissions connected to the role?

Reply:
It does now!